lighthouse-core/test/gather/gatherers/manifest-test.js

```diff
@@ -47,6 +47,46 @@ describe('Manifest gatherer', () => {
     });
   });
 
+  it('throws an error when manifest is saved as BOM UTF-8', () => {
+    const fs = require('fs');
+    const manifestWithoutBOM = fs.readFileSync(__dirname + '/../../fixtures/manifest.json')
+      .toString();
+    const manifestWithBOM = fs.readFileSync(__dirname + '/../../fixtures/manifest-bom.json')
+      .toString();
+
+    const getDriver = (data) => {
+      return {
+        driver: {
+          getAppManifest() {
+            return Promise.resolve({
+              data,
+              errors: [],
+              url: EXAMPLE_MANIFEST_URL
+            });
+          }
+        },
+        url: EXAMPLE_DOC_URL
+      };
+    };
+
+    const promises = [];
+    promises.push(manifestGather.afterPass(getDriver(manifestWithBOM))
```

Comment:
I don't think this will throw even if still bad...won't it show with an "isn't valid JSON" `debugString`?

It would be good to check the result from `afterPass`...`result.raw` should have the raw string passed in, so it should have length equal to `manifestWithBOM` minus the BOM. And it would be good to check some of the parsed values to make sure it's being parsed